### PR TITLE
Added createSourceWithParams method for iOS (#112)

### PR DESCRIPTION
**Pull request by flavordaaave**
* `STPPaymentConfiguration`: `setSmsAutofillDisabled` has been deprecated
* `PKPaymentRequest`: `paymentRequestWithMerchantIdentifier` added `country` and `currency` params
* `STPUserInformation`: `email` and `phone` have been deprecated in favor of `shippingAddress`

Regarding e2e tests with appium I got stuck when the Safari app is opened with the redirect so you need to manually authorize the transaction once the app opens the Safari browser in order to finish the test.
Here's a demo of the user flow:
ezgif-1-90721dbd68

### CODE REVIEW

**Comment by GulKhan** [comment]
Any chance you could make the same method for android as well? I think it's needed to implement a 3D Secure process.

**Comment by flavordaaave** [comment]
@GulKhan see #112 
I personally lack of android skills but @Nepherhotep mentioned they could take over the android part. (We need android as well for our alipay implementation)

**Comment by Nepherhotep** [comment]
@flavordaaave @GulKhan correct. We can start working on Android part once iOS part get reviewed and merged. It may take some time, but I don't think too much.

Thanks

**Comment by isnifer** [comment]
@flavordaaave please, rebase over master

**Comment by flavordaaave** [comment]
@isnifer done

**Comment by flavordaaave** [comment]
@dmytrozavgorodniy thanks for the detailed review. I'm really sorry for all those small "mistakes".
Should be fixed now.

File: example/src/Root.js [hunk 1]
```diff
@@ -10,12 +10,15 @@ import CardFormScreen from './scenes/CardFormScreen'
...
 import CustomBankScreen from './scenes/CustomBankScreen'
 import CardTextFieldScreen from './scenes/CardTextFieldScreen'
+import SourceScreen from './scenes/SourceScreen'
 import testID from './utils/testID'
 
 stripe.init({
   publishableKey: '<PUBLISHABLE_KEY>',
   merchantId: '<MERCHANT_ID>',
   androidPayMode: 'test',
+  country: 'US',
```
Comment (dmytrozavgorodniy) [inline example/src/Root.js]: We don't use `country`/`currency` inside `init` method in native part, so please remove it here:
```
– country: 'US',
– currency: 'usd',
```

File: example/ios/example/Info.plist [hunk 2]
```diff
@@ -20,10 +20,35 @@
...
+			<array>
+				<string>example</string>
+			</array>
+		</dict>
+	</array>
 	<key>CFBundleVersion</key>
 	<string>1</string>
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
+	<dict>
```
Comment (dmytrozavgorodniy) [inline example/ios/example/Info.plist]: Could you please use next Info.plist file. Because when you make some changes in Xcode editor, Xcode can swap some keys, but we want make minimum changes. Please use next Info.plist:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleDevelopmentRegion</key>
	<string>en</string>
	<key>CFBundleDisplayName</key>
	<string>example</string>
	<key>CFBundleExecutable</key>
	<string>$(EXECUTABLE_NAME)</string>
	<key>CFBundleIdentifier</key>
	<string>org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleName</key>
	<string>$(PRODUCT_NAME)</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleSignature</key>
	<string>????</string>
	<key>CFBundleURLTypes</key>
	<array>
		<dict>
			<key>CFBundleTypeRole</key>
			<string>Editor</string>
			<key>CFBundleURLName</key>
			<string>example</string>
			<key>CFBundleURLSchemes</key>
			<array>
				<string>example</string>
			</array>
		</dict>
	</array>
	<key>CFBundleVersion</key>
	<string>1</string>
	<key>LSRequiresIPhoneOS</key>
	<true/>
	<key>UILaunchStoryboardName</key>
	<string>LaunchScreen</string>
	<key>UIRequiredDeviceCapabilities</key>
	<array>
		<string>armv7</string>
	</array>
	<key>UISupportedInterfaceOrientations</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
		<string>UIInterfaceOrientationLandscapeLeft</string>
		<string>UIInterfaceOrientationLandscapeRight</string>
	</array>
	<key>UIViewControllerBasedStatusBarAppearance</key>
	<false/>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string></string>
	<key>NSAppTransportSecurity</key>
	<!--See http://ste.vn/2015/06/10/configuring-app-transport-security-ios-9-osx-10-11/ -->
	<dict>
		<key>NSExceptionDomains</key>
		<dict>
			<key>localhost</key>
			<dict>
				<key>NSExceptionAllowsInsecureHTTPLoads</key>
				<true/>
			</dict>
		</dict>
	</dict>
</dict>
</plist>
```

**Comment by flavordaaave** [comment]
@dmytrozavgorodniy done ;-)

**Comment by isnifer** [comment]
@flavordaaave Hi! I've switched to your fork to update tests. And I've got this:
image

What did I wrong?

**Comment by flavordaaave** [comment]
@isnifer Hmm, for some reasons the ios project in the `example_tmp` does not contain the custom URL schema setup (see: https://stripe.com/docs/mobile/ios/sources#redirecting-your-customer).

I've added this to the Xcode project in the ios folder.
You need to add:
```
#import <Stripe/Stripe.h>
```
on top of you `AppDelegate.m`

and
```
// This method handles opening native URLs (e.g., "yourexampleapp://")
- (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
  BOOL stripeHandled = [Stripe handleStripeURLCallbackWithURL:url];
  if (stripeHandled) {
    return YES;
  } else {
    // This was not a stripe url – do whatever url handling your app
    // normally does, if any.
  }
  return NO;
}
```
before `@end` statement.

Also you need to configure an custom domain for the app (the example code is using `exxample://`)
See:
bildschirmfoto 2017-09-29 um 16 54 41

**Comment by isnifer** [comment]
@flavordaaave I see everywhere on your screenshot `example`, not `exxample`. Where the mistake?

**Comment by flavordaaave** [comment]
@isnifer sorry for the typo. The screenshot shows the correct one.

**Comment by isnifer** [comment]
@flavordaaave so, I have finished with tests. For now, I need to check the new docs. But, I think you should add also `failed authorize payment` test

File: README.md [hunk 3]
```diff
@@ -574,7 +686,7 @@ An object with the following keys:
 
 ##### Example
 
-![Bank Params Android](https://cloud.githubusercontent.com/assets/1286226/26419755/6c801f06-40c9-11e7-972e-521850eda2ef.gif)
```
Comment (isnifer) [inline README.md]: @flavordaaave I think, this is the last question why did you remove this?

Comment (flavordaaave) [reply README.md]: Hmm, sorry. This should not be removed. Something went wrong here. Can you just ignore that change or should I commit an updated version where this line is till in?

Comment (isnifer) [reply README.md]: I can resume this line in your fork - no problem

**Comment by isnifer** [comment]
So, @flavordaaave I have finished work in your PR. I think tomorrow we will merge it into `master`, but we will discuss how to release it - set 4.0.0 after this PR or after Android part, cause I need to check it too.
For now, you can check if everything is OK on your side by replacing the version of `tipsi-stripe` in the `package.json` to `flavordaaave/tipsi-stripe#a11cc16`.

**Comment by isnifer** [comment]
So, @cybergrind just merge into master. Without publishing
